wolfcrypt/src/asn.c

```diff
@@ -3884,8 +3605,17 @@ int wc_EncryptPKCS8Key(byte* key, word32 keySz, byte* out, word32 outSz,
     if (ret == 0) {
         /* outer = SEQ [ pbe ] OCT encrypted_PKCS#8_key */
         outerLen = 2 + pbeLen;
-        outerLen += SetOctetString(keySz + padSz, out);
         outerLen += keySz + padSz;
+        if (out == NULL) {
+            /* plus 2 tags and 2 lengths (for sequence and octet string). */
+            *outSz = outerLen + 2 + 2*MAX_LENGTH_SZ;
```

Comment:
Can you use `SetLength(keySz + padSz) + 2`?

Reply:
Will look into it.

Reply:
I'm not sure how that would work here, since I don't want to actually touch the output buffer because it's NULL. My understanding is that `SetLength` needs a valid (i.e. non-NULL) buffer to work.